src/Model/Aggregations/Metric/GeoBounds.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace AviationCode\Elasticsearch\Model\Aggregations\Metric;
+
+use AviationCode\Elasticsearch\Helpers\HasAttributes;
+
+/**
+ * Class GeoBounds.
+ *
+ * @property array $value;
+ */
+class GeoBounds implements \JsonSerializable
+{
+    use HasAttributes;
+
+    /**
+     * GeoBounds Aggregation constructor.
+     *
+     * @param array $value
+     */
+    public function __construct(array $value)
+    {
+        $this->value = $value['value'];
```

Comment:
It actually returns `bounds` not `value` 

```json
        "<field>": {
            "bounds": {
                "top_left": {
                    "lat": 48.86111099738628,
                    "lon": 2.3269999679178
                },
                "bottom_right": {
                    "lat": 48.85999997612089,
                    "lon": 2.3363889567553997
                }
            }
        }
```

We should be able to simplify the use of the response

```php
$geoBounds->topLeft->lat;
$geoBounds->topLeft->lon;

$geoBounds->bottomRight->lat;
$geoBounds-> bottomRight->lon;
```

Reply:
You're right, I agree on this as well .. However, for `Stats` or `Extended Stats` Aggregations:
```
"<key>": {
            "count": 2,
            "min": 50.0,
            "max": 100.0,
            "avg": 75.0,
            "sum": 150.0
        }
```
```
"<key>": {
           "count": 2,
           "min": 50.0,
           "max": 100.0,
           "avg": 75.0,
           "sum": 150.0,
           "sum_of_squares": 12500.0,
           "variance": 625.0,
           "std_deviation": 25.0,
           "std_deviation_bounds": {
            "upper": 125.0,
            "lower": 25.0
           }
        }
```
Do we also need here to define a variable for each response field ? Because the current implementation has $value of type `array` in which all these response fields reside.

Reply:
The current implementation with `value` won't work since it doesn't return `value` at all. 

We should use it as a simple as possible to consume I would change it to simplify consuming part. 

In the constructor we should map the returned fields.

```php
$this->count = $value['count'];
...
```

Reply:
@Douglasdc3 Agreed , Let's go with it! I'm on it 👍

Reply:
@Douglasdc3 However, there won't be nested access, but instead array access after mapping/setting class variables, i.e.
```
$this->topLeft = $value['top_left'];
$this->bottomRight = $value['bottom_right'];
```
and from outside we'll use it as: `$geoBounds->topLeft['lat']`

OR:
```
$this->bounds = $value['bounds'];
```
from outside: `$topLeftLatitude = Arr::get($geoBounds->bounds, 'top_left.lat');`

I'd go with the first one 🤔

Reply:
@hamzahejja 

How about using a `Illuminate\Support\Fluent` class? 

```php
$this->topLeft = new Fluent($value['top_left');
$this->bottomRight = new Fluent($value['bottom_right');
```

it allows access both as array `$geoBounds->topLeft['lat']` and as property `$geoBounds->topLeft->lat`

Reply:
@Douglasdc3 Yeah using Fluent would totally solve it! I appreciate the guidance and help, I'm certainly learning along contribution in `AviationCode\elasticsearch` and yet I'm sure i have so much more to learn! With time, it'll be more easy for me to decide what and how code will fit in a specific feature now that we're getting more to know how to simplify internal implementation and end-user usage of each feature !